eclog/src/test/java/com/ericsson/bss/cassandra/ecaudit/eclog/TestLogPrinter.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecaudit.eclog;
+
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.stubbing.OngoingStubbing;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class TestLogPrinter
+{
+    @Mock
+    private PrintStream stream;
+
+    @Test(timeout = 5000)
+    public void testFiveRecords() throws UnknownHostException
+    {
+        ToolOptions options = ToolOptions.builder().build();
+        LogPrinter printer = givenPrinter(options);
+        QueueReader reader = givenRecords(false, 5, 0);
+
+        printer.print(reader);
+
+        verifySingleRecords(5);
+    }
+
+    @Test(timeout = 5000)
+    public void testFiveBatchRecords() throws UnknownHostException
+    {
+        ToolOptions options = ToolOptions.builder().build();
+        LogPrinter printer = givenPrinter(options);
+        QueueReader reader = givenRecords(true, 5, 0);
+
+        printer.print(reader);
+
+        verifyBatchRecords(5);
+    }
+
+    @Test(timeout = 5000)
+    public void testFivePlusFiveWillSkipFiveRecords() throws UnknownHostException
+    {
+        ToolOptions options = ToolOptions.builder().build();
+        LogPrinter printer = givenPrinter(options);
+        QueueReader reader = givenRecords(false, 5, 5);
+
+        printer.print(reader);
+
+        verifySingleRecords(5);
```

Comment:
Do we verify which records we skipped? Or I might be missing something from the test structure..